### DISCUSSION TITLE
Does Realtime API work with non-self-hosted apps?

### DISCUSSION THREAD

**Discussion by rogueyoshi** [supabase/supabase#1897] [Questions] [answered]
I notice I can't find a WebSocket endpoint anywhere on app.supabase.io. Does this functionality only work if you self-host?

**Comment by soedirgo** [comment, marked as answer]
Hey @rogueyoshi, the WS endpoint is at `wss://<your_supabase_ref>.supabase.co/realtime/v1/websocket` (you also need to pass an `apikey` query parameter from Settings > API > API Keys). I realize this is undocumented, sorry about that. You also need to turn on replication for tables you want to subscribe to in Database > Replication > Source.

**Reply by inian** [reply]
Yeah, we are currently not using the realtime functionality on the dashboard since it is turned off by default for new projects @rogueyoshi. We are working on something cool here though. 

It is fully supported in the hosted setup, you can still turn it on and use it in your apps as @soedirgo mentioned.

**Reply by rogueyoshi** [reply]
Thank you
